src/main/java/org/terasology/thirst/ThirstAuthoritySystem.java

```diff
@@ -42,16 +50,79 @@
  * This authority system handles drink consumption by various entities.
  */
 @RegisterSystem(value = RegisterMode.AUTHORITY)
-public class ThirstAuthoritySystem extends BaseComponentSystem {
+public class ThirstAuthoritySystem extends BaseComponentSystem implements UpdateSubscriberSystem {
     private static final Logger logger = LoggerFactory.getLogger(ThirstAuthoritySystem.class);
     @In
     private EntityManager entityManager;
     @In
     private InventoryManager inventoryManager;
     @In
+    private PrefabManager prefabManager;
+    @In
     private Time time;
+    @In
+    private DelayManager delayManager;
 
     private boolean destroyDrink = false;
+    private static long lastTick = 0;
+    /**
+     * The interval (in milliseconds) at which healthDecreaseAmount (thirstComponenent) is applied to the component.
+     */
+    private int healthDecreaseInterval = 30000;
+    public static final String THIRST_DAMAGE_ACTION_ID = "Thirst Damage";
+
+    /**
+     * Checks the ThirstComponent for all entities and triggers DamageEvents if their thirst level is below the threshold.
+     *
+     * @param delta - Unused parameter.
+     */
+    @Override
+    public void update(float delta) {
+        long gameTime = time.getGameTimeInMs();
+        if (gameTime > lastTick) {
+            lastTick = gameTime + healthDecreaseInterval;
+            for (EntityRef entity : entityManager.getEntitiesWith(ThirstComponent.class, AliveCharacterComponent.class)) {
+                ThirstComponent thirst = entity.getComponent(ThirstComponent.class);
+
+                // Check to see if health should be decreased
+                if ((ThirstUtils.getThirstForEntity(entity) < thirst.healthLossThreshold)) {
+                    if (!(delayManager.hasPeriodicAction(entity, THIRST_DAMAGE_ACTION_ID)))
+                        delayManager.addPeriodicAction(entity, THIRST_DAMAGE_ACTION_ID, 0, healthDecreaseInterval);
+                } else {
+                    if (delayManager.hasPeriodicAction(entity, THIRST_DAMAGE_ACTION_ID)) {
+                        delayManager.cancelPeriodicAction(entity, THIRST_DAMAGE_ACTION_ID);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Deals a unit of thirst damage to the character.
+     */
+    @ReceiveEvent
+    public void onPeriodicActionTriggered(PeriodicActionTriggeredEvent event, EntityRef entity) {
+        ThirstComponent thirst = entity.getComponent(ThirstComponent.class);
```

Comment:
You need to check that the I'd matches the id that you set when you start the `PeriodicAction`. Otherwise this method will get called every single time _any_ periodic action is triggered. Not just when yours is